test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpHandler.java

```diff
@@ -374,18 +428,58 @@ public void handle(final HttpExchange exchange) throws IOException {
                 }
             } else {
                 logger.warn("--> Unrecognised request received: {}", request);
-                sendError(exchange, RestStatus.BAD_REQUEST);
+                sendError(exchange, RestStatus.NOT_FOUND, "ResourceNotFound", "The specified resource doesn't exist.");
             }
+        } catch (MockAzureBlobStore.AzureBlobStoreError e) {
+            sendError(exchange, e);
+        } catch (Exception e) {
+            logger.error("Uncaught exception", e);
```

Comment:
I think this should be a test failure too

Reply:
Done in 0543928d82469192ddd7253d2fdf876b475fd619